sdk/openai/azure-ai-openai-realtime/src/main/java/com/azure/ai/openai/realtime/RealtimeAsyncClient.java

```diff
@@ -0,0 +1,480 @@
+                return Mono.error(
+                    logger.logExceptionAsError(new IllegalStateException("Failed to start. Client is not STOPPED.")));
+            } else {
+                if (postStartTask != null) {
+                    postStartTask.run();
+                }
+
+                return Mono.empty();
+            }
+        })
+            .then(authenticationProvider.authenticationToken())
+            .flatMap(authenticationHeader -> Mono.<Void>fromRunnable(() -> {
+                this.webSocketSession
+                    = webSocketClient.connectToServer(this.clientEndpointConfiguration, () -> authenticationHeader,
+                        loggerReference, this::handleMessage, this::handleSessionOpen, this::handleSessionClose);
+            }))
+            .subscribeOn(Schedulers.boundedElastic())
+            .doOnError(error -> {
+                // stop if error, do not send StoppedEvent when it fails at start(), which would have exception thrown
+                handleClientStop(false);
+            });
+    }
+
+    private Mono<Void> getStoppedByUserMono() {
+        Sinks.Empty<Void> sink = Sinks.empty();
+        boolean isStoppedByUserMonoSet = isStoppedByUserSink.compareAndSet(null, sink);
+        if (!isStoppedByUserMonoSet) {
+            sink = isStoppedByUserSink.get();
+        }
+        return sink == null ? Mono.empty() : sink.asMono();
+    }
+
+    private Mono<Void> checkStateBeforeSend() {
+        return Mono.defer(() -> {
+            RealtimeClientState state = clientState.get();
+            if (state == RealtimeClientState.CLOSED) {
+                return Mono.error(logger.logExceptionAsError(
+                    new IllegalStateException("Failed to send message. WebPubSubClient is CLOSED.")));
+            }
+            if (state != RealtimeClientState.CONNECTED) {
+                return Mono.error(
+                    logSendMessageFailedException("Failed to send message. Client is " + state.name() + ".", null,
+                        state == RealtimeClientState.RECOVERING
+                            || state == RealtimeClientState.CONNECTING
+                            || state == RealtimeClientState.RECONNECTING
+                            || state == RealtimeClientState.DISCONNECTED,
+                        (String) null));
+            }
+            if (webSocketSession == null || !webSocketSession.isOpen()) {
+                // something unexpected
+                return Mono.error(logSendMessageFailedException(
+                    "Failed to send message. Websocket session is not opened.", null, false, (String) null));
+            } else {
+                return Mono.empty();
+            }
+        });
+    }
+
+    private void handleMessage(Object message) {
+        serverEvents.tryEmitNext((RealtimeServerEvent) message);
+    }
+
+    private void handleSessionOpen(WebSocketSession session) {
+        logger.atVerbose().log("Session opened");
```

Comment:
Verbose logs should always use the supplier overload `logger.atVerbose().log(() -> "Session opened");`